### PR TITLE
fix(explore): disable resize bar when the results area is collapsed

### PR DESCRIPTION
**Pull request by stephenLYZ**
### SUMMARY

If the data section (results/samples) is collapsed, user can still drag the icon/object between chart and data section header to resize the chart. This behavior is confusing as in this case the data section remains collapsed and empty.

So we remove drag indicator/feature if the data section is collapsed.

#### before

https://user-images.githubusercontent.com/11830681/189017103-24c2c272-2fba-42dd-8b56-0fd87108395c.mov

#### after
https://user-images.githubusercontent.com/11830681/189016760-60e1de08-8d63-408a-9c8f-2903137a7366.mov

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/ExploreChartPanel.test.jsx [hunk 1]
```diff
@@ -145,4 +146,15 @@ describe('ChartContainer', () => {
...
   });
+
+  it('hide gutter when collapse panel', async () => {
+    const props = createProps();
+    setItem(LocalStorageKeys.is_datapanel_open, true);
+    const { container } = render(<ChartContainer {...props} />, {
+      useRedux: true,
+    });
+    userEvent.click(screen.getByLabelText('Collapse data panel'));
+    const gutter = container.querySelector('.gutter');
```
Comment (geido) [inline superset-frontend/src/explore/components/ExploreChartPanel.test.jsx:157]: Can you please also verify in the test that this was visible before closing the data panel?

File: superset-frontend/src/explore/components/ExploreChartPanel.test.jsx [hunk 2]
```diff
@@ -145,4 +146,15 @@ describe('ChartContainer', () => {
     render(<ChartContainer {...props} />, { useRedux: true });
     expect(screen.queryByText('Cached')).not.toBeInTheDocument();
   });
+
+  it('hide gutter when collapse panel', async () => {
```
Comment (geido) [inline superset-frontend/src/explore/components/ExploreChartPanel.test.jsx:150]:
```suggestion
  it('hides gutter when collapsing data panel', async () => {
```

**Review by geido — APPROVED** [review]
@stephenLYZ this LGTM. I am wondering if we should close the data panel automatically when it reaches the bottom, as it is in fact the same as closing it. CC @kasiazjc 

https://user-images.githubusercontent.com/60598000/189643128-0b3daa95-7e1e-44ff-8c23-6cc17cfe2058.mp4

**Comment by kasiazjc** [comment]
> @stephenLYZ this LGTM. I am wondering if we should close the data panel automatically when it reaches the bottom, as it is in fact the same as closing it. CC @kasiazjc
> 
>  Superset.mp4

good call @geido, I think it makes sense! You cannot see anything, so there is no point in it being expanded. 

There is one edge case, where someone may be playing with the drag and drop and out of the blue the section will collapse itself and they will have to manually expand which can be annoying. But I'd still say that we should go with your suggestion, Diego and we'll hear feedback that it doesn't make sense, then we'll revert it. 

cc: @stephenLYZ

**Comment by rusackas** [comment]
I feel like the suggestions being floated around might involve more edge cases and frustration than the original thing this PR fixes. Can we just merge this and ticket any additional layers of fanciness as an additional task?

**Pull request merged**